<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DDF_Editor</class>
 <widget class="QWidget" name="DDF_Editor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>610</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>1024</width>
    <height>610</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>DDF Editor</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <property name="documentMode">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="tabItems">
      <attribute name="title">
       <string>Items</string>
      </attribute>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Device</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="devManufacturerNameLabel">
           <property name="text">
            <string>Manufacturer</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="devModelIdLabel">
           <property name="text">
            <string>Model ID</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="DDF_TreeView" name="ddfTreeView">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>300</width>
             <height>0</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>16777215</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QStackedWidget" name="editStackedWidget">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>360</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <widget class="QWidget" name="editDevice">
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <property name="topMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Edit device</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QFrame" name="frame">
             <property name="frameShape">
              <enum>QFrame::StyledPanel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_6">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,0,0">
                <property name="topMargin">
                 <number>0</number>
                </property>
                <item>
                 <widget class="QLabel" name="label_12">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="text">
                   <string>Status</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QComboBox" name="devStatusComboBox">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <item>
                   <property name="text">
                    <string>Draft</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Bronze</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Silver</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Gold</string>
                   </property>
                  </item>
                 </widget>
                </item>
                <item>
                 <spacer name="horizontalSpacer">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </item>
              <item>
               <widget class="QGroupBox" name="debBasicClusterId">
                <property name="title">
                 <string>Basic cluster identification</string>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_11">
                 <item>
                  <widget class="QLabel" name="label_5">
                   <property name="text">
                    <string>Manufacturer name</string>
                   </property>
                   <property name="buddy">
                    <cstring>devManufacturerNameInput</cstring>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="TextLineEdit" name="devManufacturerNameInput">
                   <property name="toolTip">
                    <string>Manufacturer name from Basic Cluster</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="label_6">
                   <property name="text">
                    <string>Model ID</string>
                   </property>
                   <property name="buddy">
                    <cstring>devModelIdInput</cstring>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="TextLineEdit" name="devModelIdInput">
                   <property name="toolTip">
                    <string>Model ID from Basic Cluster</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <spacer name="verticalSpacer_4">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeType">
                 <enum>QSizePolicy::Preferred</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>4</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="Line" name="line">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_10">
                <property name="text">
                 <string>Vendor</string>
                </property>
                <property name="buddy">
                 <cstring>devVendorInput</cstring>
                </property>
               </widget>
              </item>
              <item>
               <widget class="TextLineEdit" name="devVendorInput">
                <property name="toolTip">
                 <string>Friendly name of the manufacturer</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_7">
                <property name="text">
                 <string>Product</string>
                </property>
                <property name="buddy">
                 <cstring>devProductInput</cstring>
                </property>
               </widget>
              </item>
              <item>
               <widget class="TextLineEdit" name="devProductInput">
                <property name="toolTip">
                 <string>Complements the model id to be shown in the UI</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QCheckBox" name="devSleeperCheckBox">
                <property name="toolTip">
                 <string>Sleeping devices can only receive when awake</string>
                </property>
                <property name="text">
                 <string>Sleeper</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="verticalSpacer_2">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="editSubdevice">
          <layout class="QVBoxLayout" name="verticalLayout_7">
           <item>
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Edit subdevice</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QFrame" name="frame_2">
             <property name="frameShape">
              <enum>QFrame::StyledPanel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_4">
              <item>
               <widget class="QLabel" name="label_8">
                <property name="text">
                 <string>Type</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="TextLineEdit" name="subDeviceTypeInput"/>
              </item>
              <item>
               <widget class="QLabel" name="label_9">
                <property name="text">
                 <string>Unique ID template</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="TextLineEdit" name="subDeviceUniqueIdInput"/>
              </item>
              <item>
               <spacer name="verticalSpacer">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>319</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="DDF_ItemEditor" name="editItem"/>
         <widget class="QWidget" name="managedItem">
          <layout class="QVBoxLayout" name="verticalLayout_9">
           <item>
            <widget class="QLabel" name="managedItemLabel">
             <property name="text">
              <string>Managed item</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QFrame" name="frame_3">
             <property name="frameShape">
              <enum>QFrame::StyledPanel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_8">
              <item>
               <widget class="QLabel" name="managedItemDescription">
                <property name="text">
                 <string>Description</string>
                </property>
                <property name="wordWrap">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="verticalSpacer_5">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeType">
                 <enum>QSizePolicy::Preferred</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>24</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QLabel" name="label_12877">
                <property name="text">
                 <string>This item is managed internally.</string>
                </property>
                <property name="wordWrap">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="verticalSpacer_3">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Available items</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="DDF_ItemList" name="itemListView">
           <property name="sizePolicy">
            <sizepolicy hsizetype="MinimumExpanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>230</width>
             <height>0</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>400</width>
             <height>16777215</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="DDF_BindingEditor" name="tabBindings">
      <attribute name="title">
       <string>Bindings</string>
      </attribute>
     </widget>
     <widget class="QWidget" name="tabPreview">
      <attribute name="title">
       <string>Preview</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_10">
       <item>
        <widget class="QPlainTextEdit" name="ddfJsonDoc">
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>DDF_TreeView</class>
   <extends>QTreeView</extends>
   <header>ui/ddf_treeview.h</header>
  </customwidget>
  <customwidget>
   <class>DDF_ItemList</class>
   <extends>QListView</extends>
   <header>ui/ddf_itemlist.h</header>
  </customwidget>
  <customwidget>
   <class>DDF_ItemEditor</class>
   <extends>QWidget</extends>
   <header>ui/ddf_itemeditor.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>TextLineEdit</class>
   <extends>QLineEdit</extends>
   <header>ui/text_lineedit.h</header>
  </customwidget>
  <customwidget>
   <class>DDF_BindingEditor</class>
   <extends>QWidget</extends>
   <header>ui/ddf_bindingeditor.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
